src/Matrix.jl

```diff
@@ -5285,7 +5285,7 @@ function snf_kb!(S::MatrixElem{T}, U::MatrixElem{T}, K::MatrixElem{T}, with_traf
                K[r, j] = reduce!(t1 + t2)
             end
          end
-         S[j, j] = divexact(S[i, i]*S[j, j], d)
+         S[j, j] = S[i, i]*divexact(S[j, j], d)
```

Comment:
Just to explain, this tries to compute "the" `lcm` of two values `a` and `b` where `d` is the gcd, by doing `(a*b)/d`, which I change to `a*(b/d)`. 

This makes a difference if there are zero divisors, e.g. in $\mathbb{Z}/8\mathbb{Z}$ if $a=b=4$ then $d=4$, then $(a\cdot b)/d = 0$, while $a\cdot(b/d) = 4$.